src/libraries/System.Console/src/System/ConsolePal.iOS.cs

```diff
@@ -0,0 +1,189 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Text;
+
+namespace System
+{
+    internal sealed class NSLogStream : Stream
+    {
+        public override void Flush() { }
+
+        public override int Read(byte[] buffer, int offset, int count) => throw new NotSupportedException();
+
+        public override long Seek(long offset, SeekOrigin origin) => throw new NotSupportedException();
+
+        public override void SetLength(long value) => throw new NotSupportedException();
+
+        public override unsafe void Write(byte[] buffer, int offset, int count)
+        {
+            if (count > 0 && count <= buffer.Length - offset)
+            {
+                fixed (byte* ptr = buffer)
+                {
+                    Interop.Sys.Log(ptr + offset, count);
+                }
+            }
+        }
+
+        public override bool CanRead => false;
+
+        public override bool CanSeek => false;
+
+        public override bool CanWrite => true;
+
+        public override long Length => throw new NotSupportedException();
+
+        public override long Position
+        {
+            get => throw new NotSupportedException();
+            set => throw new NotSupportedException();
+        }
+    }
+
+    internal static class ConsolePal
+    {
+        public static Stream OpenStandardInput() => throw new PlatformNotSupportedException();
+
+        public static Stream OpenStandardOutput() => new NSLogStream();
+
+        public static Stream OpenStandardError() => new NSLogStream();
+
+        public static Encoding InputEncoding => throw new PlatformNotSupportedException();
+
+        public static void SetConsoleInputEncoding(Encoding enc) => throw new PlatformNotSupportedException();
+
+        public static Encoding OutputEncoding => Encoding.Unicode;
+
+        // underlying API expects only utf-16
```

Comment:
`SystemNative_Log` expects utf8, not utf16.

Reply:
@rolfbjarne does it? https://github.com/dotnet/runtime/blob/1ef7abff2e0e1bd8c392cce88f29b5fe9bf1d13d/src/libraries/Native/Unix/System.Native/pal_log.m#L10
I could remove it and do Unicode->Utf8 in C# but I suspect it's faster to do it in the native code.

Reply:
Ah, nvm, I misread the code.